### PR TITLE
🌱 drop the draft release notes step from github workflow

### PR DESCRIPTION
**Pull request by ykakarap**
**What this PR does / why we need it**:

Since the release team is taking care of generating the release notes as needed, dropping the step from the github workflow.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by ykakarap** [comment]
cc @kubernetes-sigs/cluster-api-release-team

**Review by cpanato — APPROVED** [review]
/lgtm

**Comment by hackeramitkumar** [comment]
+1
/lgtm

**Review by furkatgofurov7 — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -27,12 +27,9 @@ jobs:
...
-      - name: generate release notes
-        run: |
-          make release-notes
       - name: Release
         uses: softprops/action-gh-release@de2c0eb89ae2a093876385947365aca7b0e5f844 # tag=v1
         with:
           draft: true
           files: out/*
-          body_path: _releasenotes/${{ env.RELEASE_TAG }}.md
+          body_path: "TODO: Copy release notes shared by the comms team"
```
Comment (sbueringer) [inline .github/workflows/release.yml:35]:
```suggestion
          body: "TODO: Copy release notes shared by the comms team"
```

Comment (ykakarap) [reply .github/workflows/release.yml:35]: Good catch!

**Comment by sbueringer** [comment]
/lgtm
/approve

I guess this PR is enough to change the workflow for all branches 

Let's test it with the upcoming patch releases

**Comment by ykakarap** [comment]
>I guess this PR is enough to change the workflow for all branches

I believe so. Since the workflow is triggered with a tag push and a tag push is not any branch specific, as far as git is considered, I believe the default branch will be used for the workflow.

We will know for sure during the next patch releases 😅

**Comment by sbueringer** [comment]
Looks like it didn't work for v1.3.3

Let's backport

**Pull request merged**